### ISSUE TITLE
Difference in outputs from cross_validation function between daily and weekly data

### ISSUE THREAD

**Issue by plbalmeida** [facebook/prophet#2005]
Hi all!

I noticed that the metrics results from the _cross_validation_ and _performance_metrics_ functions were not varying depending on the combination of hyperparameters. I'm coding in pyspark and using applyInPandas [()](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.GroupedData.applyInPandas.html) to forecast a **weekly** time series in pyspark data frame format.

So I got an example from medium [()](https://higee.medium.com/parallel-model-training-with-prophet-and-spark-5f40be750f97) with a **daily** time series and the outputs from _cross_validation_ and _performance_metrics_ functions are varying.

![image](https://user-images.githubusercontent.com/46826549/130310325-18bae0a4-9a24-4818-9647-f1dd1f35a42b.png)

For a test, I resampled the same data from tutorial to weekly timestamp, and the outputs did not vary.   

![image](https://user-images.githubusercontent.com/46826549/130310334-e5e3ed12-9758-4163-90ba-39517927df8a.png)

The function that I'm using to run cross validation using prophet is:

`
# based on: https://higee.medium.com/parallel-model-training-with-prophet-and-spark-5f40be750f97
def run_cross_validation(keys, df):
  
  # sorting through timestamp
  df = df.sort_values('ds')
  
  # setting keys
  item, seasonality_mode, changepoint_prior_scale, seasonality_prior_scale = keys
  
  # model instance
  model = Prophet(
    seasonality_mode=seasonality_mode, 
    changepoint_prior_scale=changepoint_prior_scale, 
    seasonality_prior_scale=seasonality_prior_scale
  )
  
  # training model
  model.fit(df)

  # cross-validation for daily data
  df_cv = cross_validation(
    model,
    horizon=pd.Timedelta('{} W'.format(8)),
    parallel='processes'
  )
  
  # getting performance metrics 
  df_perf_metrics = performance_metrics(
    df_cv,
    metrics=['mse'],
    rolling_window=1
  )

  metric = np.mean(df_perf_metrics['mse'].values[:3])

  df_perf_metrics_mse = pd.DataFrame(
    data=[metric],
    columns=['mse']
  )

  mapping = {
    'item': item,
    'seasonality_mode': seasonality_mode,
    'changepoint_prior_scale': changepoint_prior_scale,
    'seasonality_prior_scale': seasonality_prior_scale
  }

  df_perf_metrics_mse = df_perf_metrics_mse.assign(**mapping)

  cols = [
    'item',
    'mse', 
    'seasonality_mode', 
    'changepoint_prior_scale',
    'seasonality_prior_scale'
  ]

  return df_perf_metrics_mse[cols]
`

Does anyone know what might be wrong with what I'm doing?
